creator-node/src/services/prometheusMonitoring/prometheusMetrics.constants.js

```diff
@@ -0,0 +1,54 @@
+const promClient = require('prom-client')
+
+const MetricTypes = Object.freeze({
+  COUNTER: promClient.Counter,
+  GAUGE: promClient.Gauge,
+  HISTOGRAM: promClient.Histogram,
+  SUMMARY: promClient.Summary
```

Comment:
I marked this thread as unresolved so this comment wouldn't get lost. Feel free to resolve again once you've processed it!

* https://prometheus.io/docs/practices/instrumentation/#counter-vs-gauge-summary-vs-histogram

Most notably:

> Raw counters are rarely useful. Use the rate() function to get the per-second rate at which they are increasing.

and

> You should never take a rate() of a gauge.


Personally, it seems like I’ve always focused on this part: `Raw counters are rarely useful.` But since [I don’t fully understand](https://audius-internal.slack.com/archives/CA80RCL77/p1653612781935569?thread_ts=1653601873.494019&cid=CA80RCL77) the PromQL syntax at times, I’ve kept it simple for myself: use histograms for almost all cases since they export a [*_count metric](https://discoveryprovider2.staging.audius.co/prometheus_metrics) (when I will want to use `rate()` which is included in the majority of the PromQL queries I copy-paste, or someone with more skills will want to run an advanced query later) or use gauges for all other point-in-time metrics (if I needed to use `rate()` I would have been using a histogram).

Reply:
chatted on call, am convinced to remove for now - histograms are better